### PR TITLE
feat(keys): add init code

### PR DESCRIPTION
**Pull request by vladikoff**
Connects to https://github.com/mozilla/fxa/issues/245

### CODE REVIEW

File: app/scripts/models/auth_brokers/oauth.js [hunk 1]
```diff
@@ -81,17 +82,95 @@ define(function (require, exports, module) {
...
+        })
+        .then((rkeys) => {
+          var uid = account.get('uid');
+          return this.relier.deriveRelierKeys(rkeys, uid);
+        })
+        .then((rkeys) => {
+          keys = rkeys;
+          var pk = JSON.parse(relier.get('jwk'))
+
+          return window.crypto.subtle.importKey(
```
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/oauth.js]: Do we have to worry about older browsers that don't support window.crypto?

File: app/scripts/models/auth_brokers/oauth.js [hunk 2]
```diff
@@ -81,17 +82,95 @@ define(function (require, exports, module) {
...
+            },
+            false, //whether the key is extractable (i.e. can be used in exportKey)
+            ["encrypt"] //"encrypt" or "wrapKey" for public key import or
+            //"decrypt" or "unwrapKey" for private key imports
+          )
+
+        })
+        .then((pk) => {
+          function str2ab(str) {
+            var buf = new ArrayBuffer(str.length*2); // 2 bytes for each char
```
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/oauth.js]: And ArrayBuffer/Uint16Array?

File: app/scripts/lib/crypto/base64url.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Base64url encoding/decoding of SJCL bitArrays.
+ *
+ * SJCL natively supports standard base64 encoding but not the urlsafe
+ * variant, which is needed for JWTs.
+ */
+
+define([
+  'sjcl'
```
Comment (shane-tomlinson) [inline app/scripts/lib/crypto/base64url.js]: I know it's early in the dev cycle still, though I wonder if sjcl can be loaded using requireOnDemand. IIRC, sjcl is pretty hefty.

File: app/scripts/models/auth_brokers/oauth.js (hunk 1 above)
Comment (vladikoff) [inline app/scripts/models/auth_brokers/oauth.js]: Nah, I think we are going for Firefox 57 for now (in the next iteration Android 4+)

File: app/scripts/lib/crypto/base64url.js (hunk 3 above)
Comment (vladikoff) [inline app/scripts/lib/crypto/base64url.js]: That might just work!

**Pull request closed without merging**